### PR TITLE
Azure: use release-1.11 branch for CAPZ v1beta1

### PR DESCRIPTION
**Pull request by nojnhuh**
Updates references to CAPZ's release-1.10 branch to release-1.11.

/hold for the CAPZ branch to be cut.

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.10.yaml
Comment (nojnhuh) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.10.yaml:1]: The way I updated these 1.10 and 1.11 files was roughly:
- rename 1.10 -> 1.11
- change branch references in 1.11 to release-1.11
- rename 1.9 -> 1.10
- change branch references in 1.10 to release-1.10
- add back missing Workload Identity creds to 1.10 jobs

Comment (mboersma) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.10.yaml:1]: That should do it, sorry it's so fiddly.

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.10.yaml
Comment (nojnhuh) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.10.yaml:1]: Thanks! I can handle fiddly, just wanted to make sure I'm fiddling sensibly 😄

**Comment by nojnhuh** [comment]
Branch has been cut! https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/release-1.11

/hold cancel
/assign @andyzhangx @feiskyer @marosset

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Comment by nojnhuh** [comment]
@andyzhangx @feiskyer PTAL

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**